lib/spack/spack/ci/__init__.py

```diff
@@ -637,13 +641,25 @@ def download_and_extract_artifacts(url, work_dir):
         response = urlopen(request, timeout=SPACK_CDASH_TIMEOUT)
         with open(artifacts_zip_path, "wb") as out_file:
             shutil.copyfileobj(response, out_file)
+
+        with zipfile.ZipFile(artifacts_zip_path) as zip_file:
+            zip_file.extractall(work_dir)
+            # Get the artifact root
+            artifact_root = ""
+            for f in zip_file.filelist:
+                if "spack.lock" in f.filename:
+                    artifact_root = os.path.dirname(os.path.dirname(f.filename))
```

Comment:
It was a pain to trace through from here looking for where this value is born, I'm sorry we both had to do that :laughing:  I know there is still lots to do to make the ci modules nicer to work with.